plugins/quota/actor.go

```diff
@@ -0,0 +1,100 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+// isQuotadAffectedNode returns true if there are local bricks of volume on
+// which quota is enabled
+func isQuotadAffectedNode(volumes []*volume.Volinfo) bool {
+	for _, v := range volumes {
+		if !isQuotaEnabled(v) {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			}
+			continue
+		}
+	}
+	return false
+}
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key, value string, logger log.FieldLogger) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+	// Create pidfile dir if not exists
+	if err := os.MkdirAll(path.Dir(quotadDaemon.pidfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	// Create logFiledir dir
+	if err := os.MkdirAll(path.Dir(quotadDaemon.logfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	volumes, err := volume.GetVolumes()
+	if err != nil {
+		logger.WithError(err).Error("failed to get volumes")
+		return err
+	}
+	// Check for making changes to quotad on this node
+	if !isQuotadAffectedNode(volumes) {
+		// This condition is for disable
+		// If its the last node that is going to stop the daemon,
+		// it enters the if part.
+		err = daemon.Stop(quotadDaemon, true, logger)
+		if err != nil {
+			logger.Error("quotad stop failed")
+		}
+	} else {
+		// This is to handle the all the start and also the stop
+		// is there is another volume with quota enabled on the same
+		// node
+		err = daemon.Stop(quotadDaemon, true, logger)
+		if err == errors.ErrPidFileNotFound {
+			logger.Info("quotad stop failed as pidfile missing")
+		} else {
```

Comment:
I think you missed handling err==nil part

Reply:
done